asv_bench/benchmarks/sparse.py

```diff
@@ -1,211 +1,162 @@
-        self.b = self.make_sparse_array(length=1000000, num_blocks=1000,
-                                        block_size=10, fill_value=np.nan)
-
-        self.a_zero = self.make_sparse_array(length=1000000, num_blocks=1000,
-                                             block_size=10, fill_value=0)
-        self.b_zero = self.make_sparse_array(length=1000000, num_blocks=1000,
-                                             block_size=10, fill_value=np.nan)
+    def setup(self, dense_proportion, fill_value):
+        N = 10**6
+        arr1 = make_array(N, dense_proportion, fill_value, np.int64)
+        self.array1 = SparseArray(arr1, fill_value=fill_value)
+        arr2 = make_array(N, dense_proportion, fill_value, np.int64)
+        self.array2 = SparseArray(arr2, fill_value=fill_value)
 
-    def make_sparse_array(self, length, num_blocks, block_size, fill_value):
-        a = np.array([fill_value] * length)
-        for block in range(num_blocks):
-            i = np.random.randint(0, length)
-            a[i:i + block_size] = np.random.randint(0, 100, len(a[i:i + block_size]))
-        return pd.SparseArray(a, fill_value=fill_value)
+    def time_make_union(self, dense_proportion, fill_value):
+        self.array1.sp_index.make_union(self.array2.sp_index)
 
-    def time_sparse_make_union(self):
-        self.a.sp_index.make_union(self.b.sp_index)
+    def time_intersect(self, dense_proportion, fill_value):
+        self.array1.sp_index.intersect(self.array2.sp_index)
 
-    def time_sparse_intersect(self):
-        self.a.sp_index.intersect(self.b.sp_index)
+    def time_add(self, dense_proportion, fill_value):
+        self.array1 + self.array2
 
-    def time_sparse_addition(self):
-        self.a + self.b
+    def time_divide(self, dense_proportion, fill_value):
+        self.array1 / self.array2
 
-    def time_sparse_addition_zero(self):
-        self.a_zero + self.b_zero
 
-    def time_sparse_division(self):
-        self.a / self.b
+class ArithmeticBlock(object):
 
-    def time_sparse_division_zero(self):
-        self.a_zero / self.b_zero
+    goal_time = 0.2
+    params = [np.nan, 0]
+    param_names = ['fill_value']
+
+    def setup(self, fill_value):
+        N = 10**6
+        self.arr1 = self.make_block_array(length=N, num_blocks=1000,
+                                          block_size=10, fill_value=fill_value)
+        self.arr2 = self.make_block_array(length=N, num_blocks=1000,
+                                          block_size=10, fill_value=fill_value)
+
+    def make_block_array(self, length, num_blocks, block_size, fill_value):
+        arr = np.full(length, fill_value)
+        indicies = np.random.choice(np.arange(0, length, block_size),
+                                    num_blocks,
+                                    replace=False)
+        for ind in indicies:
+            arr[ind:ind + block_size] = np.random.randint(0, 100, block_size)
+        return SparseArray(arr, fill_value=fill_value)
```

Comment:
@mroeschke looking at this now, can we turn the Arithmetic class above into a special case of ArithmeticBlock?

Reply:
Sure I suppose so, though I didn't write the original benchmarks.